packages/cardano-services/src/WsServer/transactions.ts

```diff
@@ -0,0 +1,728 @@
+        stakeAddress
+      }))
+    : undefined;
+
+const signatures = new Map<Ed25519PublicKeyHex, Ed25519SignatureHex>();
+
+// eslint-disable-next-line complexity
+const mapTx = (tx: TxModel): Cardano.HydratedTx => {
+  const collateralReturn = tx.collateral.address ? mapOutput(tx.collateral) : undefined;
+  const collaterals = tx.collaterals[0].txId ? tx.collaterals : [];
+  const fee = BigInt(tx.fee);
+  const inputs = tx.inputs[0].address ? tx.inputs : [];
+  const outputs = tx.outputs[0].address ? tx.outputs.map(mapOutput) : [];
+
+  return {
+    // Let's do this forcing to perform the deserialization on client side
+    auxiliaryData: tx.metadata[0][1] ? (tx.metadata as unknown as Cardano.AuxiliaryData) : undefined,
+    blockHeader: { blockNo: tx.block, hash: tx.hash, slot: tx.slot },
+    body: {
+      ...(tx.valid
+        ? { collateralReturn, collaterals, fee, inputs, outputs }
+        : {
+            collateralReturn: outputs[0],
+            collaterals: inputs,
+            fee: 0n,
+            inputs: [],
+            outputs: [],
+            totalCollateral: fee
+          }),
+      certificates: mapCertificates(tx),
+      mint: tx.mint[0][0] ? new Map(tx.mint.map(mapAsset)) : undefined,
+      proposalProcedures: mapActions(tx.actions),
+      validityInterval: { invalidBefore: tx.before || undefined, invalidHereafter: tx.hereafter || undefined },
+      votingProcedures: tx.votes[0].role ? mapVotes(tx.votes) : undefined,
+      withdrawals: mapWithdrawals(tx.withdrawals)
+    },
+    id: tx.id,
+    index: tx.index,
+    inputSource: tx.valid ? Cardano.InputSource.inputs : Cardano.InputSource.collaterals,
+    txSize: tx.size,
+    witness: { redeemers: tx.redeemers[0].purpose ? tx.redeemers.map(mapRedeemer) : undefined, signatures }
+  };
+};
+
+const actions = [{}] as ActionModel[];
+const collateral = {} as OutputModel;
+const collaterals = [{}] as Cardano.HydratedTxIn[];
+const commReg = [[]] as unknown as TxModel['commReg'];
+const commRet = [[]] as unknown as TxModel['commRet'];
+const deRep = [[]] as unknown as TxModel['deRep'];
+const metadata = [[]] as unknown as MetadataDbModel[];
+const mint = [[]] as unknown as TxModel['mint'];
+const redeemers = [{}] as RedeemerModel[];
+const reg = [[]] as unknown as TxModel['reg'];
+const sDele = [[]] as unknown as TxModel['sDele'];
+const spReg = [[]] as unknown as TxModel['spReg'];
+const spRetire = [[]] as unknown as TxModel['spRetire'];
+const unReg = [[]] as unknown as TxModel['unReg'];
+const vDele = [[null]] as unknown as TxModel['vDele'];
+const votes = [{}] as VoteModel[];
+const withdrawals = [{}] as TxModel['withdrawals'];
+
+export const transactionsByIds = async (ids: string[], db: Pool, onlyUtxos?: boolean) => {
+  const result = await db.query<{ tx: TxModel }>({
+    ...(onlyUtxos ? { name: 'get_utxos', text: getUtxos } : { name: 'get_txs', text: getTransactions }),
+    values: [ids]
+  });
+
+  const rows = onlyUtxos
```

Comment:
for utxo this probalby should be a separate funciton

Reply:
8d0f6acf25e should solve this